provider/test-programs/changing-region/index.ts

```diff
@@ -0,0 +1,14 @@
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+
+const cfg = new pulumi.Config();
+const reg: aws.Region = cfg.require("desired-region");
+
+const awsProvider = new aws.Provider("myprovider", {
```

Comment:
Just curious would this work any differently with default providers?

Reply:
I don't think there's any difference but I can add a test.